torch_geometric/data/lightning_datamodule.py

```diff
@@ -331,7 +333,13 @@ def __repr__(self) -> str:
 ###############################################################################
 
 
-def infer_input_nodes(data: Union[Data, HeteroData], split: str) -> InputNodes:
+def infer_input_nodes(data: Union[Data, HeteroData, Tuple[FeatureStore,
+                                                          GraphStore]],
+                      split: str) -> InputNodes:
+    if isinstance(data, Tuple):
+        # The GraphStore is the HeteroData at the moment.
+        data = data[1]
```

Comment:
I am pretty sure this will not be sufficient for general `GraphStore` objects, and will only work for `HeteroData` objects masquerading as `GraphStores. `GraphStore` does not define `__contains__`, so checking whether certain attributes are in `data` is confusing. Furthermore, in the current definition of `GraphStore`, we only store edge indices in the graph store, so masks and indices would logically be placed in the feature store, not the graph store.

In order to fix this properly, I'd suggest obtaining `feature_store, graph_store = data` and checking for the `attr_name`s in `feature_store` with the feature store API calls.